### DISCUSSION TITLE
requested features?

### DISCUSSION THREAD

**Discussion by preetamslot** [unfolding-io/nebulix#14] [General]
Hi Astronauts,
I am thrilled with all the positive feedback! so thanks for that!  
If you have feature requests, please let me know, this will help allot!

**Comment by jpollone** [comment]
No, thank you, @preetamslot! I have a few thoughts and some likely not "features" but things I notice. 😁

1. One item for you to understand is the amount of time you saved me. I appreciate it and hope I can return the favor in the future. How about a "Made with Unfolding.io/Nebulix" (a Showcase page) where we can list our public sites to show support for your work?
2. On project/work, I can't seem to get #more working with a Split page. I really like your split page idea where I list features (such as skills and client), but I do need a button for the user to click to view more. (Full page works fine with #more.)
3. It looks like I'll need to write a few case studies for a few of these jobs I'm applying for. I'll keep a copy of the study in my space because they take a loooooong time to put together, so why not? I'd like more separation like what you did for Roorda with Werk and Cases. Is there an easier way to implement these separate "entities"?
4. I notice the text after the first paragraph is reduced by a certain amount. Can we override this from the StaticCMS implementation?
5. One feature might be to look at the additional page animation since we can fun things now with Astro. I am thinking about something like a Figma Prototype where I have a case the user is reading, then they click a button and a new page slides in from the right until the show is over. That's not a lot of detail for you, but being able to click through a full-page frame in a Figma Prototype to transition to another page always stood out.

**Reply by jpollone** [reply]
Oh! I just noticed after this saved, there are more discussions. I can break out these ideas tomorrow if that makes more sense.

**Reply by preetamslot** [reply]
@jpollone Thank for the feedback.
I will add a section "made with Nebulix" when there are live sites, great idea!

About the more link, at this moment it points to the "page builder block section" 
If you add an `id="features"` to the feature section in `src/pages/work[...slug].astro` then you can create a scroll link to that with `#features`

**EDIT:**
If the collection is the same in structure, you could also add a a field to the collection like "type".
then filter the type in the `src/pages/work` and  `src/pages/cases`


**You can create more content collections:**
For the cms part look at this folder https://github.com/unfolding-io/nebulix/tree/main/src/cms
save the `projects.mjs` file as `case.mjs`, and rename the export `export const case=` and change the folder `folder: "src/content/case"` 
and edit the tags field:
```
 {
      name: "tag",
      label: t("tags"),
      multiple: true,
      widget: "relation",
      collection: "case",
      max: 5,
      required: false,
      file: "project",
      search_fields: ["case_tags.*.name"],
      display_fields: ["case_tags.*.name"],
      value_field: "case_tags.*.name",
    },
```
Import and export it in the `/src/cms/index.mjs`
you can go to the `/src/pages/admin.astro` and add the collection to the imports and config of the CMS.

You would need to create a settings file for this collection:
https://github.com/unfolding-io/nebulix/blob/main/src/cms/settings.mjs
Just duplicate the project object and modify some fields

```
...
 {
      name: "case",
      label: t("case_settings"),
      file: "src/content/config/case.mdx",

      fields: [
        { name: "title", label: t("title"), widget: "string" },
    
        ....

        {
          name: "case_tags",
          ...
        },
       ...
      ],
    },
```
Create the case config in the CMS and add a case.

**Add the collection to Astro:**
Go to https://github.com/unfolding-io/nebulix/blob/main/src/content/config.ts#L134

and duplicate the project collection, and rename it to case, change the work_tags to case_tags. and add case to the exports at the bottom of the page.

Then copy de `src/pages/work` folder and rename it to case.
now open the `src/pages/case/[...page]` & `src/pages/case/[...slug]` and replace the `"project"` with`"case"` 
and change the base_url to case  `base_url = "case"`

if all goes wel this should work!.

**About the first paragraph**
this is pure css, you can remove it here:
https://github.com/unfolding-io/nebulix/blob/main/src/styles/global.css#L236
you can also just add the class 'no-lead' to a page builder block to disable this style.

**About the page animations.**
That is a great idea, have to think about how this will work. for now we have the morph animation as default and fade if no animation name is found. I could make it so that you choose  the default animation (slide: left, right up, down). great idea!

Thanks again

**Reply by jpollone** [reply]
Thank you for the detail, @preetamslot. I'll make the smaller tweaks this week, but will likely transform Cases the next week since it's probably more important for me to focus on adding more portfolio items. This way, when I send potential employers to the portfolio, I can refer to the page if I have follow-up call.

**Comment by getofmeland** [comment]
Is tit possible to look at an integration with Sendgrid and other email tools?

Also would like to look to move my deployment over to Azure Static Web Apps.

**Reply by getofmeland** [reply]
Also integration with other newsletter tools

**Reply by preetamslot** [reply]
Hi @getofmeland 
Here an example for a sendgrid edge-function:
https://github.com/unfolding-io/StarFunnel/blob/master/edge-functions/contact-mailgun/contact-mailgun.js

I am working on an update and m thinking to make use of Astro endpoints for he forms, so then the deployment method would become hybrid.

What email tools are you using, I have looked t active campaign, but had not found "the best way" for integration.

**Reply by getofmeland** [reply]
Oh I did look there but that appears to still be mailgun? or can it be used for both?

Newsletter tools, I was looking at substack to see if there was some kind of API integration

**Reply by preetamslot** [reply]
my bad!
I have not made sendgrid implementation for these themes.
I can work on it. but do you want to use sendgrid for contact form or newsletters?

**Comment by getofmeland** [comment]
Another feature request would be the ability to add an RSS feed

**Reply by getofmeland** [reply]
Create new file in src/pages/rss.xml.js

With the following code:

````
import rss, { pagesGlobToRssItems } from '@astrojs/rss';
import { getCollection } from 'astro:content';

export async function GET(context) {
  const posts = await getCollection('blog');
  return rss({
    title: 'ENTER BLOG NAME | Blog',
    description: 'Enter description',
    // Pull in your project "site" from the endpoint context
    // https://docs.astro.build/en/reference/api-reference/#contextsite
    site: context.site,
    items: posts.map((post) => ({
      // Assumes all RSS feed item properties are in post frontmatter
      ...post.data,
      // Generate a `url` from each post `slug`
      // This assumes all blog posts are rendered as `/blog/[slug]` routes
      // https://docs.astro.build/en/guides/content-collections/#generating-pages-from-content-collections
      link: `/blog/${post.slug}/`,
    })),
  });
}
````

**Reply by preetamslot** [reply]
I think RSS is great, but this also should be done for other content collections, I will have to look into this. thanks

**Comment by getofmeland** [comment]
Is there any option for getting analytics for page views etc?  Thinking from a blog perspective or a google analytic type scenario

**Reply by preetamslot** [reply]
tbh, I dont like GA, but here re alternatives like: https://plausible.io/ 
This is easy to implement.

**Reply by getofmeland** [reply]
How easy would this be to implement, and submit the sitemap to search engines etc

**Comment by getofmeland** [comment]
Another thought was the ability to do a drop down menu on the Main Menu, not sure if this is something that's possible

**Reply by preetamslot** [reply]
This can be done, do you have an example on design (mobile & desktop). 
Let me know and ill put it in the backlog.
